bundles/org.openhab.binding.webexteams/src/main/java/org/openhab/binding/webexteams/internal/WebexTeamsHandler.java

```diff
@@ -0,0 +1,429 @@
+            throw new WebexTeamsException("OAuth exception", e);
+        }
+    }
+
+    public boolean isAuthorized() {
+        final AccessTokenResponse accessTokenResponse = getAccessTokenResponse();
+
+        if ("person".equals(this.accountType)) {
+            return accessTokenResponse != null && accessTokenResponse.getAccessToken() != null
+                    && accessTokenResponse.getRefreshToken() != null;
+        } else {
+            // bots don't need no refreshToken!
+            return accessTokenResponse != null && accessTokenResponse.getAccessToken() != null;
+        }
+    }
+
+    private @Nullable AccessTokenResponse getAccessTokenResponse() {
+        try {
+            return this.authService == null ? null : this.authService.getAccessTokenResponse();
+        } catch (OAuthException | IOException | OAuthResponseException | RuntimeException e) {
+            logger.debug("Exception checking authorization: ", e);
+            return null;
+        }
+    }
+
+    public boolean equalsThingUID(String thingUID) {
+        return getThing().getUID().getAsString().equals(thingUID);
+    }
+
+    public String formatAuthorizationUrl(String redirectUri) {
+        try {
+            if (this.configured) {
+                return this.authService.getAuthorizationUrl(redirectUri, null, thing.getUID().getAsString());
+            } else {
+                return "";
+            }
+        } catch (final OAuthException e) {
+            logger.debug("Error constructing AuthorizationUrl: ", e);
+            return "";
+        }
+    }
+
+    // mainly used to refresh the auth token when using OAuth
+    private boolean refresh() {
+        synchronized (refreshSynchronization) {
+            Person person;
+            try {
+                person = client.getPerson();
+                String type = person.getType();
+                if (type == null) {
+                    type = "?";
+                }
+                updateProperty(PROPERTY_WEBEX_TYPE, type);
+                this.accountType = type;
+                updateProperty(PROPERTY_WEBEX_NAME, person.getDisplayName());
+
+                // Only when the identity is a person:
+                if ("person".equalsIgnoreCase(person.getType())) {
+                    String status = person.getStatus();
+                    updateState(CHANNEL_STATUS, StringType.valueOf(status));
+                    DateFormat df = new SimpleDateFormat(ISO8601_FORMAT);
+                    String lastActivity = df.format(person.getLastActivity());
+                    updateState(CHANNEL_LASTACTIVITY, new DateTimeType(lastActivity));
+                }
+                updateStatus(ThingStatus.ONLINE);
+                return true;
+            } catch (WebexTeamsException e) {
+                logger.warn("Failed to refresh: {}", e.getMessage());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
```

Comment:
Maybe rather COMMUNICATION_ERROR ?
It could fail for another reason than a bad configuration.

Reply:
Updated.